LanguageFeatures/Patterns/type_inference_A13_t03.dart

```diff
@@ -29,7 +27,4 @@ main() {
     dynamic e = 2.71;
     var (int v2) = e;
   });
-
-  final (int Function(int) v3) = foo;
-  v3.expectStaticType<Exactly<int Function(int)>>();
```

Comment:
This one should continue to work: It is not a coercion during pattern matching, it is a coercion that is applied to the initializing expression based on the context type schema.

This is just another case where the already existing coercion mechanism kicks in ("the initializing expression doesn't care where that context type schema came from").